[0.18.0] build failure with old GCC

On RHEL / CentOS (GCC 4)

```
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c: In function 'mpack_str_check_no_null':
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c:849:5: error: 'for' loop initial declarations are only allowed in C99 mode
BUILDSTDERR:      for (size_t i = 0; i < bytes; ++i)
BUILDSTDERR:      ^
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c:849:5: note: use option -std=c99 or -std=gnu99 to compile your code
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c: In function 'mpack_expect_enum':
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c:4019:5: error: 'for' loop initial declarations are only allowed in C99 mode
BUILDSTDERR:      for (size_t i = 0; i < count; ++i) {
BUILDSTDERR:      ^
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c: In function 'mpack_expect_enum_optional':
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c:4052:5: error: 'for' loop initial declarations are only allowed in C99 mode
BUILDSTDERR:      for (size_t i = 0; i < count; ++i) {
BUILDSTDERR:      ^
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c: In function 'mpack_node_map_int_impl':
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c:5829:5: error: 'for' loop initial declarations are only allowed in C99 mode
BUILDSTDERR:      for (size_t i = 0; i < node.data->len; ++i) {
BUILDSTDERR:      ^
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c: In function 'mpack_node_map_uint_impl':
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c:5860:5: error: 'for' loop initial declarations are only allowed in C99 mode
BUILDSTDERR:      for (size_t i = 0; i < node.data->len; ++i) {
BUILDSTDERR:      ^
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c: In function 'mpack_node_map_str_impl':
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c:5894:5: error: 'for' loop initial declarations are only allowed in C99 mode
BUILDSTDERR:      for (size_t i = 0; i < node.data->len; ++i) {
BUILDSTDERR:      ^
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c: In function 'mpack_node_enum_optional':
BUILDSTDERR: /builddir/build/BUILD/php-pecl-datadog-trace-0.18.0/NTS/src/ext/mpack/mpack.c:5996:5: error: 'for' loop initial declarations are only allowed in C99 mode
BUILDSTDERR:      for (size_t i = 0; i < count; ++i) {
BUILDSTDERR:      ^
BUILDSTDERR: make: *** [src/ext/mpack/mpack.lo] Error 1
BUILDSTDERR: make: *** Waiting for unfinished jobs....

```

--- a/config.m4
+++ b/config.m4
@@ -10,7 +10,7 @@ if test "$PHP_DDTRACE" != "no"; then
     PHP_SUBST(EXTRA_CFLAGS)
   fi
 
-  PHP_NEW_EXTENSION(ddtrace, src/ext/ddtrace.c src/ext/dispatch_setup.c src/ext/dispatch.c src/ext/request_hooks.c src/ext/compat_zend_string.c src/ext/dispatch_compat_php5.c src/ext/dispatch_compat_php7.c src/ext/backtrace.c src/ext/logging.c src/ext/env_config.c src/ext/serializer.c src/ext/mpack/mpack.c, $ext_shared,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
+  PHP_NEW_EXTENSION(ddtrace, src/ext/ddtrace.c src/ext/dispatch_setup.c src/ext/dispatch.c src/ext/request_hooks.c src/ext/compat_zend_string.c src/ext/dispatch_compat_php5.c src/ext/dispatch_compat_php7.c src/ext/backtrace.c src/ext/logging.c src/ext/env_config.c src/ext/serializer.c src/ext/mpack/mpack.c, $ext_shared,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -Wall -std=c99)
   PHP_ADD_BUILD_DIR($ext_builddir/src/ext, 1)
 
   PHP_ADD_INCLUDE($ext_builddir)

